silkrpc/consensus/ethash.cpp

```diff
@@ -16,15 +16,43 @@
 
 #include "ethash.hpp"
 
+#include <silkworm/core/silkworm/chain/protocol_param.hpp>
+#include <silkrpc/json/types.hpp>
+
 namespace silkrpc::ethash {
 
 BlockReward compute_reward(const ChainConfig& config, const silkworm::Block& block) {
-    return BlockReward{};
+    auto chain_config = silkworm::ChainConfig::from_json(config.config).value();
+    auto revision = chain_config.revision(block.header.number);
+    BlockReward block_reward;
+    block_reward.miner_reward = silkworm::param::kBlockRewardFrontier;
+    if (revision > evmc_revision::EVMC_BYZANTIUM) {
+        block_reward.miner_reward = silkworm::param::kBlockRewardByzantium;
+    }
+    if (revision > evmc_revision::EVMC_CONSTANTINOPLE) {
+        block_reward.miner_reward = silkworm::param::kBlockRewardConstantinople;
+    }
+
+    // Accumulate the rewards for the miner and any included uncles
+    intx::uint256 ommer_reward;
+    auto header_number = block.header.number;
+    for (auto uncle : block.ommers) {
```

Comment:
Probably `uncle` can be declared `const auto&`

Reply:
Done